### PR TITLE
<feat> test suite

### PR DESCRIPTION
**Pull request by CodeNoobKing**
## Summary by CodeRabbit

- **New Features**
	- Introduced a comprehensive testing suite for SOFAArk functionality, including class loading, plugin registration, and test execution.
	- Added Spring configuration setups for testing within a Spring context.
	- Implemented Maven project configuration for dependency management and plugin setup.
- **Tests**
	- Added various mock classes and tests to validate the functionality of the test suite, including compatibility tests with JUnit4 and multi-application setups in Spring.
- **Documentation**
	- Updated `MANIFEST.MF` and test configuration files to support new testing capabilities.

### CODE REVIEW

**Comment by CodeNoobKing** [comment]
@coderabbitai review

File: koupleless-ext/koupleless-test-suite/src/main/java/com/alipay/sofa/koupleless/test/suite/biz/SOFAArkTestClassLoaderHook.java [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+import java.util.List;
+import java.util.concurrent.ConcurrentMap;
+import java.util.jar.JarEntry;
+import java.util.jar.JarFile;
+
+/**
+ * @author CodeNoobKing
+ * @date 2024/3/21
+ **/
+public class SOFAArkTestClassLoaderHook extends DelegateToMasterBizClassLoaderHook {
```
Comment (lvjing2) [inline koupleless-ext/koupleless-test-suite/src/main/java/com/alipay/sofa/koupleless/test/suite/biz/SOFAArkTestClassLoaderHook.java:42]: 可以不用加 SOFAArk 前缀吧

File: pom.xml [hunk 2]
```diff
@@ -179,6 +179,12 @@
                 <version>${hessian.version}</version>
             </dependency>
 
+            <dependency>
+                <groupId>org.projectlombok</groupId>
+                <artifactId>lombok</artifactId>
+                <version>1.18.30</version>
```
Comment (lvjing2) [inline pom.xml:185]: 版本号要统一放在跟 pom 里定义

Comment (CodeNoobKing) [reply pom.xml:196]: 这个文件是根 POM

**Review by lvjing2 — APPROVED** [review]
LGTM

**Pull request merged**